frontends/p4/structInitializers.cpp

```diff
@@ -134,4 +322,44 @@ const IR::Node* CreateStructInitializers::postorder(IR::Operation_Relation* expr
     return expression;
 }
 
+const IR::Node* CreateStructAssignInitializers::postorder(IR::AssignmentStatement* statement) {
```

Comment:
will your code handle structure expressions that are not used just as initializers?
e.g. `if (x == { a = 2, ... })`?

Reply:
if not, we should make sure that the spec does not allow this.

Reply:
Currently that use case is not covered at all, I made an omission. Do we want to support those use cases, or compiler should report an error?